src/appleseed-max-impl/builtinmapsupport.cpp

```diff
@@ -43,12 +44,41 @@
 namespace asf = foundation;
 namespace asr = renderer;
 
+void connect_output_selector(
+    asr::ShaderGroup&   shader_group,
+    const char*         material_node_name,
+    const char*         material_input_name,
+    Texmap*             texmap)
+{
+    const auto t = GetCOREInterface()->GetTime();
+
+    Texmap* input_map = nullptr;
+    get_paramblock_value_by_name(texmap->GetParamBlock(0), L"source_map", t, input_map, FOREVER);
+    int output_index = 0;
+    get_paramblock_value_by_name(texmap->GetParamBlock(0), L"output_index", t, output_index, FOREVER);
+
+    if (input_map != nullptr && is_osl_texture(input_map))
+    {
+        OSLTexture* osl_texture = static_cast<OSLTexture*>(input_map);
+        auto output_names = osl_texture->get_output_names();
+
+        if (--output_index < output_names.size())
```

Comment:
That doesn't look correct. What if `output_index` is 0?

Reply:
That parameter should not be less than 1 but I agree, it's not safe.
Would this kind of check work? `if (output_index > 0 && --output_index < output_names.size())`